tests/foreman/cli/test_remoteexecution.py

```diff
@@ -1019,3 +1020,204 @@ def test_positive_rex_against_infra_hosts(
+    def test_positive_run_job_on_host_converted_to_pull_provider(
+        self,
+        module_org,
+        smart_proxy_location,
+        module_ak_with_cv,
+        module_capsule_configured_mqtt,
+        rhel_contenthost,
+    ):
+        """Run custom template on host converted to mqtt
+
+        :id: 9ad68172-de7b-4578-a3ae-49b6ff461691
+
+        :expectedresults: Verify the job was successfully ran against the host converted to mqtt
+
+        :CaseImportance: Critical
+
+        :parametrized: yes
+        """
+        result = rhel_contenthost.execute(
+            f'curl -o /etc/pki/ca-trust/source/anchors/satellite-sat-engineering-ca.crt \
+                    {settings.repos["DOGFOOD_REPO_HOST"]}/pub/katello-server-ca.crt; \
+                    update-ca-trust'
+        )
+        assert result.status == 0, 'Failed to download certificate'
+        client_repo = (
+            f'{settings.repos["DOGFOOD_REPO_HOST"].replace("http", "https")}/pulp/content/'
+            'Satellite_Engineering/QA/Satellite_Client/custom/Satellite_Client_Composes/'
+            f'Satellite_Client_RHEL{rhel_contenthost.os_version.major}_x86_64/'
+        )
+        # TODO client_repo should be changed to
+        # settings.repos['SATCLIENT_REPO'][f'RHEL{rhel_contenthost.os_version.major}']
+        # when/if the new dogfood url pattern settles
+
+        # register host with rex, enable client repo, install katello-agent
+        result = rhel_contenthost.register(
+            module_capsule_configured_mqtt,
+            module_org,
+            smart_proxy_location,
+            module_ak_with_cv.name,
+            packages=['katello-agent'],
+            repo=client_repo,
+        )
+        assert result.status == 0, f'Failed to register host: {result.stderr}'
+        # result = rhel_contenthost.execute('subsription-manager identity/status')
+
+        # install conversion script (SAT-1670)
+        result = rhel_contenthost.execute('yum install -y katello-pull-transport-migrate')
+        assert result.status == 0, 'Failed to install katello-pull-transport-migrate'
+        # check mqtt client is running
+        result = rhel_contenthost.execute('yggdrasil status')
+        assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+        result = rhel_contenthost.execute('systemctl status yggdrasild')
+        assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+        # run script provider rex command
+        invocation_command = make_job_invocation(
+            {
+                'job-template': 'Run Command - Script Default',
+                'inputs': 'command=ls',
+                'search-query': f"name ~ {rhel_contenthost.hostname}",
+            }
+        )
+        result = JobInvocation.info({'id': invocation_command['id']})
+        try:
+            assert result['success'] == '1'
+        except AssertionError:
+            result = 'host output: {}'.format(
```

Comment:
golfing a bit here, but no need to label `result` here, just call AssertionError and define the string in it's call parenthesis.

Reply:
submitted https://github.com/SatelliteQE/robottelo/issues/9985 for this